swim/memberlist.go

```diff
@@ -275,16 +278,138 @@ func (m *memberlist) MakeFaulty(address string, incarnation int64) []Change {
 }
 
 func (m *memberlist) SetLocalStatus(status string) {
+	m.members.Lock()
 	m.local.Status = status
+	m.members.Unlock()
 	m.postLocalUpdate()
 }
 
+func (m *memberlist) SetLocalLabel(key, value string) error {
```

Comment:
I had thought about it, but was worried about garbage creation which is not really relevant since it is not expected to be called so often that it would matter.

I can refactor this code to use `SetLocalLabels` with a 1-element-map